### PR TITLE
Extract _build_row_lengths_offsets function

### PR DESCRIPTION
**Pull request by ericbn**

### CODE REVIEW

File: src/ssort/_parsing.py [hunk 1]
```diff
@@ -106,53 +107,13 @@ def split_class(statement):
...
-
-    if node.body[0].lineno == token.end[0]:
+    _, row_offsets = _build_row_lengths_offsets(text_padded)
+
+    if node.lineno == node.body[0].lineno:
         # All tokens are on the same line.  `split` won't know how to indent
         # them so we do it ourselves.
-        head_end_lineno, head_end_col = token.end
+        head_end_lineno = node.lineno
+        head_end_col = node.body[0].col_offset - 1
```
Comment (ericbn) [inline src/ssort/_parsing.py:116]: `- 1` is enough here since an `rstrip()` will be done below.

**Review by bwhmather — COMMENTED** [review]
Sorry for delay.  Have difficulty getting free time when I also have internet access.

After much puzzling, I think I've figured out what on earth I was thinking with the original implementation.

I think the missing test case is:
```python
def test_split_class_with_bases():
    actual = _split_class("class A(\n    B,\n):pass")
    expected = "class A(\n    B,\n):", ["    pass"]
    assert actual == expected
```

Here the line with `class...` on it is not the same as the line that ends the class head.

With this change we get:
```python
'class A(', ['    B,\n):pass']
```
Sadly I don't think there is a node for the trailing bracket or colon so I don't know what can be done.

**Comment by ericbn** [comment]
I totally appreciate your patience in reviewing this and not letting me almost mess it up!  :- )

I've updated the PR with a smaller refactoring, adding the test case that you mentioned and also fixing the recent error with the coverage check.

File: tests/test_ast.py [hunk 2]
```diff
@@ -30,6 +30,10 @@
 def _nodes_types(
     node_type: type[ast.AST] = ast.AST,
 ) -> Iterable[type[ast.AST]]:
+    # coverage package adds a coverage.parser.NodeList subclass
```
Comment (bwhmather) [inline tests/test_ast.py:33]: That's just naughty.  Why on earth does it do that?

**Comment by ericbn** [comment]
> Thanks. Mind restoring the two test cases you had in before? They looked worthwhile.

Sure, done!

**Review by bwhmather — APPROVED** [review]
Thank you very much.  Very happy to take any other improvements that might tickle your fancy.

**Pull request merged**